pymodbus/client/base.py

```diff
@@ -53,20 +53,20 @@ class ModbusBaseClient(ModbusClientMixin, ModbusProtocol):
     class _params:
         """Parameter class."""
 
-        retries: int = None
-        retry_on_empty: bool = None
-        close_comm_on_error: bool = None
-        strict: bool = None
-        broadcast_enable: bool = None
-        reconnect_delay: int = None
+        retries: int | None = None
+        retry_on_empty: bool | None = None
+        close_comm_on_error: bool | None = None
+        strict: bool | None = None
+        broadcast_enable: bool | None = None
+        reconnect_delay: int | None = None
 
-        source_address: tuple[str, int] = None
+        source_address: tuple[str, int] | None = None
 
-        server_hostname: str = None
+        server_hostname: str | None = None
 
     def __init__(  # pylint: disable=too-many-arguments
         self,
-        framer: type[ModbusFramer] = None,
+        framer: type[ModbusFramer],
```

Comment:
Well, there is a type error in `__init__` if this stays optional which should then be resolved in another way. How to initialize `self.framer` if the argument to `framer` not given and thus `framer` defaults to None?

Reply:
That is another problem, which I think is handled at lower levels in the code.

Please do not change API, because this will cause every app to be changed.

Reply:
Apart from that the type error is quite easy to solve without changing the parameter itself.

Reply:
Ok, I did not see how to resolve that easily. Here I need your support.